### ISSUE TITLE
gevent 1.1b2, PyPy, requests, OS X: DisutilsModuleError: can't compile C/C++ code

### ISSUE THREAD

**Issue by rouge8** [gevent/gevent#631]
When running this script:

```python
import gevent.monkey
gevent.monkey.patch_all()

import requests
print requests.get('https://www.google.com')
```

with PyPy 2.6:

```console
$ python --version
Python 2.7.9 (295ee98b69288471b0fcf2e0ede82ce5209eb90b, Jun 02 2015, 18:26:45)
[PyPy 2.6.0 with GCC 4.2.1 Compatible Apple LLVM 6.1.0 (clang-602.0.53)]
```

and Gevent 1.1b2:

```console
$ pip freeze -l
gevent==1.1b2
requests==2.7.0
```

I get a `DistutilsModuleError`.

```pytb
$ python wat.py
Traceback (most recent call last):
  File "<builtin>/app_main.py", line 75, in run_toplevel
  File "wat.py", line 5, in <module>
    print requests.get('https://www.google.com')
  File "/Users/rouge8/.virtualenvs/tmp-17448bae0f2be4dc/site-packages/requests/api.py", line 69, in get
    return request('get', url, params=params, **kwargs)
  File "/Users/rouge8/.virtualenvs/tmp-17448bae0f2be4dc/site-packages/requests/api.py", line 49, in request
    session = sessions.Session()
  File "/Users/rouge8/.virtualenvs/tmp-17448bae0f2be4dc/site-packages/requests/sessions.py", line 291, in __init__
    self.headers = default_headers()
  File "/Users/rouge8/.virtualenvs/tmp-17448bae0f2be4dc/site-packages/requests/utils.py", line 573, in default_headers
    'User-Agent': default_user_agent(),
  File "/Users/rouge8/.virtualenvs/tmp-17448bae0f2be4dc/site-packages/requests/utils.py", line 560, in default_user_agent
    p_system = platform.system()
  File "/usr/local/Cellar/pypy/2.6.0_1/libexec/lib-python/2.7/platform.py", line 1310, in system
    return uname()[0]
  File "/usr/local/Cellar/pypy/2.6.0_1/libexec/lib-python/2.7/platform.py", line 1277, in uname
    processor = _syscmd_uname('-p','')
  File "/usr/local/Cellar/pypy/2.6.0_1/libexec/lib-python/2.7/platform.py", line 1008, in _syscmd_uname
    f = os.popen('uname %s 2> %s' % (option, DEV_NULL))
DistutilsModuleError: can't compile C/C++ code: unable to load module 'distutils.unixccompiler'
```

**Comment by jamadden** [comment]
What about the versions of gcc/clang that are on your path?

**Comment by jamadden** [comment]
Assuming that the developer tools are installed on the path and up-to-date, we've also seen this same (confusing) error for invalid environment variables like CFLAGS and LDFLAGS so you might remove anything you have set there too.

**Comment by rouge8** [comment]
```console
$ gcc --version
Configured with: --prefix=/Library/Developer/CommandLineTools/usr --with-gxx-include-dir=/usr/include/c++/4.2.1
Apple LLVM version 6.1.0 (clang-602.0.53) (based on LLVM 3.6.0svn)
Target: x86_64-apple-darwin14.4.0
Thread model: posix
$ clang --version
Apple LLVM version 6.1.0 (clang-602.0.53) (based on LLVM 3.6.0svn)
Target: x86_64-apple-darwin14.4.0
Thread model: posix
$ env | grep FLAGS
XPC_FLAGS=0x0
```

Same error after `unset`ing `XPC_FLAGS`. Not sure what that does, but I figured unsetting wouldn't hurt.

**Comment by jamadden** [comment]
Ok, just some guesses.

The traceback there isn't very helpful (it's the traceback from requests, not the underlying traceback that comes from compiling the CFFI module). You should get a slightly more helpful traceback by putting this at the very top of your file:

```
from gevent.core import loop
```

That should force immediate compilation and present the raw traceback.

**Comment by rouge8** [comment]
That actually fixes it entirely. And weirdly, that fix persists in the virtualenv even if I uninstall and reinstall gevent and remove `from gevent.core import loop`, i.e.

```sh
virtualenv gevent-631 -p /usr/local/bin/pypy
source gevent-631/bin/activate
pip install gevent==1.1b2 requests
python -c "from gevent.core import loop"
python ~/Desktop/631.py
pip uninstall -y gevent
pip install gevent==1.1b2
python ~/Desktop/631.py
cat ~/Desktop/631.py
```

On the other hand, if I create a virtualenv and don't run  `python -c "from gevent.core import loop"`, it fails like in the first traceback. Full console output below:

```console
$ virtualenv gevent-631 -p /usr/local/bin/pypy
Running virtualenv with interpreter /usr/local/bin/pypy
New pypy executable in gevent-631/bin/pypy
Installing setuptools, pip, wheel...done.
$ source gevent-631/bin/activate
$ pip install gevent==1.1b2 requests
Collecting gevent==1.1b2
Collecting requests
  Using cached requests-2.7.0-py2.py3-none-any.whl
Installing collected packages: gevent, requests
Successfully installed gevent-1.1b2 requests-2.7.0
$ python -c "from gevent.core import loop"
In file included from gevent-631/site-packages/gevent/__pycache__/_cffi__gd8408504xde6c861b.c:54:
In file included from /Users/rouge8/tmp/gevent-631/site-packages/gevent/libev.h:2:
/Users/rouge8/tmp/gevent-631/site-packages/gevent/libev/ev.c:483:48: warning: '/*' within block comment
      [-Wcomment]
/*#define MIN_INTERVAL  0.00000095367431640625 /* 1/2**20, good till 2200 */
                                               ^
/Users/rouge8/tmp/gevent-631/site-packages/gevent/libev/ev.c:1692:31: warning: 'extern' variable has an
      initializer [-Wextern-initializer]
  EV_API_DECL struct ev_loop *ev_default_loop_ptr = 0; /* needs to be initialised to make it a d...
                              ^
In file included from gevent-631/site-packages/gevent/__pycache__/_cffi__gd8408504xde6c861b.c:54:
In file included from /Users/rouge8/tmp/gevent-631/site-packages/gevent/libev.h:2:
In file included from /Users/rouge8/tmp/gevent-631/site-packages/gevent/libev/ev.c:2557:
/Users/rouge8/tmp/gevent-631/site-packages/gevent/libev/ev_poll.c:110:18: warning: expression result
      unused [-Wunused-value]
        assert (("libev: poll() returned illegal result, broken BSD kernel?", p < polls + pollcnt));
                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/assert.h:93:25: note: expanded from macro 'assert'
    (__builtin_expect(!(e), 0) ? __assert_rtn(__func__, __FILE__, __LINE__, #e) : (void)0)
                        ^
In file included from gevent-631/site-packages/gevent/__pycache__/_cffi__gd8408504xde6c861b.c:54:
In file included from /Users/rouge8/tmp/gevent-631/site-packages/gevent/libev.h:2:
/Users/rouge8/tmp/gevent-631/site-packages/gevent/libev/ev.c:2963:12: warning: expression result unused
      [-Wunused-value]
  assert (("libev: watcher has invalid priority", ABSPRI (w) >= 0 && ABSPRI (w) < NUMPRI));
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/assert.h:93:25: note: expanded from macro 'assert'
    (__builtin_expect(!(e), 0) ? __assert_rtn(__func__, __FILE__, __LINE__, #e) : (void)0)
                        ^
In file included from gevent-631/site-packages/gevent/__pycache__/_cffi__gd8408504xde6c861b.c:54:
In file included from /Users/rouge8/tmp/gevent-631/site-packages/gevent/libev.h:2:
/Users/rouge8/tmp/gevent-631/site-packages/gevent/libev/ev.c:2966:14: warning: expression result unused
      [-Wunused-value]
    assert (("libev: pending watcher not on pending queue", pendings [ABSPRI (w)][w->pending - 1...
             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/assert.h:93:25: note: expanded from macro 'assert'
    (__builtin_expect(!(e), 0) ? __assert_rtn(__func__, __FILE__, __LINE__, #e) : (void)0)
                        ^
In file included from gevent-631/site-packages/gevent/__pycache__/_cffi__gd8408504xde6c861b.c:54:
In file included from /Users/rouge8/tmp/gevent-631/site-packages/gevent/libev.h:2:
/Users/rouge8/tmp/gevent-631/site-packages/gevent/libev/ev.c:2976:16: warning: expression result unused
      [-Wunused-value]
      assert (("libev: active index mismatch in heap", ev_active (ANHE_w (heap [i])) == i));
               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/assert.h:93:25: note: expanded from macro 'assert'
    (__builtin_expect(!(e), 0) ? __assert_rtn(__func__, __FILE__, __LINE__, #e) : (void)0)
                        ^
In file included from gevent-631/site-packages/gevent/__pycache__/_cffi__gd8408504xde6c861b.c:54:
In file included from /Users/rouge8/tmp/gevent-631/site-packages/gevent/libev.h:2:
/Users/rouge8/tmp/gevent-631/site-packages/gevent/libev/ev.c:2977:16: warning: expression result unused
      [-Wunused-value]
      assert (("libev: heap condition violated", i == HEAP0 || ANHE_at (heap [HPARENT (i)]) <= A...
               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/assert.h:93:25: note: expanded from macro 'assert'
    (__builtin_expect(!(e), 0) ? __assert_rtn(__func__, __FILE__, __LINE__, #e) : (void)0)
                        ^
In file included from gevent-631/site-packages/gevent/__pycache__/_cffi__gd8408504xde6c861b.c:54:
In file included from /Users/rouge8/tmp/gevent-631/site-packages/gevent/libev.h:2:
/Users/rouge8/tmp/gevent-631/site-packages/gevent/libev/ev.c:2978:16: warning: expression result unused
      [-Wunused-value]
      assert (("libev: heap at cache mismatch", ANHE_at (heap [i]) == ev_at (ANHE_w (heap [i]))));
               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/assert.h:93:25: note: expanded from macro 'assert'
    (__builtin_expect(!(e), 0) ? __assert_rtn(__func__, __FILE__, __LINE__, #e) : (void)0)
                        ^
In file included from gevent-631/site-packages/gevent/__pycache__/_cffi__gd8408504xde6c861b.c:54:
In file included from /Users/rouge8/tmp/gevent-631/site-packages/gevent/libev.h:2:
/Users/rouge8/tmp/gevent-631/site-packages/gevent/libev/ev.c:2989:16: warning: expression result unused
      [-Wunused-value]
      assert (("libev: active index mismatch", ev_active (ws [cnt]) == cnt + 1));
               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/assert.h:93:25: note: expanded from macro 'assert'
    (__builtin_expect(!(e), 0) ? __assert_rtn(__func__, __FILE__, __LINE__, #e) : (void)0)
                        ^
In file included from gevent-631/site-packages/gevent/__pycache__/_cffi__gd8408504xde6c861b.c:54:
In file included from /Users/rouge8/tmp/gevent-631/site-packages/gevent/libev.h:2:
/Users/rouge8/tmp/gevent-631/site-packages/gevent/libev/ev.c:3007:14: warning: expression result unused
      [-Wunused-value]
    assert (("libev: negative fd in fdchanges", fdchanges [i] >= 0));
             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/assert.h:93:25: note: expanded from macro 'assert'
    (__builtin_expect(!(e), 0) ? __assert_rtn(__func__, __FILE__, __LINE__, #e) : (void)0)
                        ^
In file included from gevent-631/site-packages/gevent/__pycache__/_cffi__gd8408504xde6c861b.c:54:
In file included from /Users/rouge8/tmp/gevent-631/site-packages/gevent/libev.h:2:
/Users/rouge8/tmp/gevent-631/site-packages/gevent/libev/ev.c:3020:24: warning: expression result unused
      [-Wunused-value]
              assert (("libev: io watcher list contains a loop", w != w2));
                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/assert.h:93:25: note: expanded from macro 'assert'
    (__builtin_expect(!(e), 0) ? __assert_rtn(__func__, __FILE__, __LINE__, #e) : (void)0)
                        ^
In file included from gevent-631/site-packages/gevent/__pycache__/_cffi__gd8408504xde6c861b.c:54:
In file included from /Users/rouge8/tmp/gevent-631/site-packages/gevent/libev.h:2:
/Users/rouge8/tmp/gevent-631/site-packages/gevent/libev/ev.c:3024:20: warning: expression result unused
      [-Wunused-value]
          assert (("libev: inactive fd watcher on anfd list", ev_active (w) == 1));
                   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/assert.h:93:25: note: expanded from macro 'assert'
    (__builtin_expect(!(e), 0) ? __assert_rtn(__func__, __FILE__, __LINE__, #e) : (void)0)
                        ^
In file included from gevent-631/site-packages/gevent/__pycache__/_cffi__gd8408504xde6c861b.c:54:
In file included from /Users/rouge8/tmp/gevent-631/site-packages/gevent/libev.h:2:
/Users/rouge8/tmp/gevent-631/site-packages/gevent/libev/ev.c:3025:20: warning: expression result unused
      [-Wunused-value]
          assert (("libev: fd mismatch between watcher and anfd", ((ev_io *)w)->fd == i));
                   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/assert.h:93:25: note: expanded from macro 'assert'
    (__builtin_expect(!(e), 0) ? __assert_rtn(__func__, __FILE__, __LINE__, #e) : (void)0)
                        ^
In file included from gevent-631/site-packages/gevent/__pycache__/_cffi__gd8408504xde6c861b.c:54:
In file included from /Users/rouge8/tmp/gevent-631/site-packages/gevent/libev.h:2:
/Users/rouge8/tmp/gevent-631/site-packages/gevent/libev/ev.c:3207:24: warning: expression result unused
      [-Wunused-value]
              assert (("libev: negative ev_timer repeat value found while processing timers", w-...
                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/assert.h:93:25: note: expanded from macro 'assert'
    (__builtin_expect(!(e), 0) ? __assert_rtn(__func__, __FILE__, __LINE__, #e) : (void)0)
                        ^
In file included from gevent-631/site-packages/gevent/__pycache__/_cffi__gd8408504xde6c861b.c:54:
In file included from /Users/rouge8/tmp/gevent-631/site-packages/gevent/libev.h:2:
/Users/rouge8/tmp/gevent-631/site-packages/gevent/libev/ev.c:3269:24: warning: expression result unused
      [-Wunused-value]
              assert (("libev: ev_periodic reschedule callback returned time in the past", ev_a...
                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/assert.h:93:25: note: expanded from macro 'assert'
    (__builtin_expect(!(e), 0) ? __assert_rtn(__func__, __FILE__, __LINE__, #e) : (void)0)
                        ^
In file included from gevent-631/site-packages/gevent/__pycache__/_cffi__gd8408504xde6c861b.c:54:
In file included from /Users/rouge8/tmp/gevent-631/site-packages/gevent/libev.h:2:
/Users/rouge8/tmp/gevent-631/site-packages/gevent/libev/ev.c:3408:12: warning: expression result unused
      [-Wunused-value]
  assert (("libev: ev_loop recursion during release detected", loop_done != EVBREAK_RECURSE));
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/assert.h:93:25: note: expanded from macro 'assert'
    (__builtin_expect(!(e), 0) ? __assert_rtn(__func__, __FILE__, __LINE__, #e) : (void)0)
                        ^
In file included from gevent-631/site-packages/gevent/__pycache__/_cffi__gd8408504xde6c861b.c:54:
In file included from /Users/rouge8/tmp/gevent-631/site-packages/gevent/libev.h:2:
/Users/rouge8/tmp/gevent-631/site-packages/gevent/libev/ev.c:3529:22: warning: expression result unused
      [-Wunused-value]
            assert (("libev: pipe_w not active, but pipe not written", ev_is_active (&pipe_w)));
                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/assert.h:93:25: note: expanded from macro 'assert'
    (__builtin_expect(!(e), 0) ? __assert_rtn(__func__, __FILE__, __LINE__, #e) : (void)0)
                        ^
In file included from gevent-631/site-packages/gevent/__pycache__/_cffi__gd8408504xde6c861b.c:54:
In file included from /Users/rouge8/tmp/gevent-631/site-packages/gevent/libev.h:2:
/Users/rouge8/tmp/gevent-631/site-packages/gevent/libev/ev.c:3703:12: warning: expression result unused
      [-Wunused-value]
  assert (("libev: ev_io_start called with negative fd", fd >= 0));
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/assert.h:93:25: note: expanded from macro 'assert'
    (__builtin_expect(!(e), 0) ? __assert_rtn(__func__, __FILE__, __LINE__, #e) : (void)0)
                        ^
In file included from gevent-631/site-packages/gevent/__pycache__/_cffi__gd8408504xde6c861b.c:54:
In file included from /Users/rouge8/tmp/gevent-631/site-packages/gevent/libev.h:2:
/Users/rouge8/tmp/gevent-631/site-packages/gevent/libev/ev.c:3704:12: warning: expression result unused
      [-Wunused-value]
  assert (("libev: ev_io_start called with illegal event mask", !(w->events & ~(EV__IOFDSET | EV...
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/assert.h:93:25: note: expanded from macro 'assert'
    (__builtin_expect(!(e), 0) ? __assert_rtn(__func__, __FILE__, __LINE__, #e) : (void)0)
                        ^
In file included from gevent-631/site-packages/gevent/__pycache__/_cffi__gd8408504xde6c861b.c:54:
In file included from /Users/rouge8/tmp/gevent-631/site-packages/gevent/libev.h:2:
/Users/rouge8/tmp/gevent-631/site-packages/gevent/libev/ev.c:3713:12: warning: expression result unused
      [-Wunused-value]
  assert (("libev: ev_io_start called with corrupted watcher", ((WL)w)->next != (WL)w));
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/assert.h:93:25: note: expanded from macro 'assert'
    (__builtin_expect(!(e), 0) ? __assert_rtn(__func__, __FILE__, __LINE__, #e) : (void)0)
                        ^
In file included from gevent-631/site-packages/gevent/__pycache__/_cffi__gd8408504xde6c861b.c:54:
In file included from /Users/rouge8/tmp/gevent-631/site-packages/gevent/libev.h:2:
/Users/rouge8/tmp/gevent-631/site-packages/gevent/libev/ev.c:3715:34: warning: '&' within '|'
      [-Wbitwise-op-parentheses]
  fd_change (EV_A_ fd, w->events & EV__IOFDSET | EV_ANFD_REIFY);
                       ~~~~~~~~~~^~~~~~~~~~~~~ ~
/Users/rouge8/tmp/gevent-631/site-packages/gevent/libev/ev.c:3715:34: note: place parentheses around the
      '&' expression to silence this warning
  fd_change (EV_A_ fd, w->events & EV__IOFDSET | EV_ANFD_REIFY);
                                 ^
                       (                      )
/Users/rouge8/tmp/gevent-631/site-packages/gevent/libev/ev.c:3728:12: warning: expression result unused
      [-Wunused-value]
  assert (("libev: ev_io_stop called with illegal fd (must stay constant after start!)", w->fd >...
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/assert.h:93:25: note: expanded from macro 'assert'
    (__builtin_expect(!(e), 0) ? __assert_rtn(__func__, __FILE__, __LINE__, #e) : (void)0)
                        ^
In file included from gevent-631/site-packages/gevent/__pycache__/_cffi__gd8408504xde6c861b.c:54:
In file included from /Users/rouge8/tmp/gevent-631/site-packages/gevent/libev.h:2:
/Users/rouge8/tmp/gevent-631/site-packages/gevent/libev/ev.c:3748:12: warning: expression result unused
      [-Wunused-value]
  assert (("libev: ev_timer_start called with negative timer repeat value", w->repeat >= 0.));
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/assert.h:93:25: note: expanded from macro 'assert'
    (__builtin_expect(!(e), 0) ? __assert_rtn(__func__, __FILE__, __LINE__, #e) : (void)0)
                        ^
In file included from gevent-631/site-packages/gevent/__pycache__/_cffi__gd8408504xde6c861b.c:54:
In file included from /Users/rouge8/tmp/gevent-631/site-packages/gevent/libev.h:2:
/Users/rouge8/tmp/gevent-631/site-packages/gevent/libev/ev.c:3776:14: warning: expression result unused
      [-Wunused-value]
    assert (("libev: internal timer heap corruption", ANHE_w (timers [active]) == (WT)w));
             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/assert.h:93:25: note: expanded from macro 'assert'
    (__builtin_expect(!(e), 0) ? __assert_rtn(__func__, __FILE__, __LINE__, #e) : (void)0)
                        ^
In file included from gevent-631/site-packages/gevent/__pycache__/_cffi__gd8408504xde6c861b.c:54:
In file included from /Users/rouge8/tmp/gevent-631/site-packages/gevent/libev.h:2:
/Users/rouge8/tmp/gevent-631/site-packages/gevent/libev/ev.c:3838:16: warning: expression result unused
      [-Wunused-value]
      assert (("libev: ev_periodic_start called with negative interval value", w->interval >= 0.));
               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/assert.h:93:25: note: expanded from macro 'assert'
    (__builtin_expect(!(e), 0) ? __assert_rtn(__func__, __FILE__, __LINE__, #e) : (void)0)
                        ^
In file included from gevent-631/site-packages/gevent/__pycache__/_cffi__gd8408504xde6c861b.c:54:
In file included from /Users/rouge8/tmp/gevent-631/site-packages/gevent/libev.h:2:
/Users/rouge8/tmp/gevent-631/site-packages/gevent/libev/ev.c:3870:14: warning: expression result unused
      [-Wunused-value]
    assert (("libev: internal periodic heap corruption", ANHE_w (periodics [active]) == (WT)w));
             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/assert.h:93:25: note: expanded from macro 'assert'
    (__builtin_expect(!(e), 0) ? __assert_rtn(__func__, __FILE__, __LINE__, #e) : (void)0)
                        ^
In file included from gevent-631/site-packages/gevent/__pycache__/_cffi__gd8408504xde6c861b.c:54:
In file included from /Users/rouge8/tmp/gevent-631/site-packages/gevent/libev.h:2:
/Users/rouge8/tmp/gevent-631/site-packages/gevent/libev/ev.c:3907:12: warning: expression result unused
      [-Wunused-value]
  assert (("libev: ev_signal_start called with illegal signal number", w->signum > 0 && w->signu...
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/assert.h:93:25: note: expanded from macro 'assert'
    (__builtin_expect(!(e), 0) ? __assert_rtn(__func__, __FILE__, __LINE__, #e) : (void)0)
                        ^
In file included from gevent-631/site-packages/gevent/__pycache__/_cffi__gd8408504xde6c861b.c:54:
In file included from /Users/rouge8/tmp/gevent-631/site-packages/gevent/libev.h:2:
/Users/rouge8/tmp/gevent-631/site-packages/gevent/libev/ev.c:3910:12: warning: expression result unused
      [-Wunused-value]
  assert (("libev: a signal must not be attached to two different loops",
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/assert.h:93:25: note: expanded from macro 'assert'
    (__builtin_expect(!(e), 0) ? __assert_rtn(__func__, __FILE__, __LINE__, #e) : (void)0)
                        ^
In file included from gevent-631/site-packages/gevent/__pycache__/_cffi__gd8408504xde6c861b.c:54:
In file included from /Users/rouge8/tmp/gevent-631/site-packages/gevent/libev.h:2:
/Users/rouge8/tmp/gevent-631/site-packages/gevent/libev/ev.c:4028:12: warning: expression result unused
      [-Wunused-value]
  assert (("libev: child watchers are only supported in the default loop", loop == ev_default_lo...
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/assert.h:93:25: note: expanded from macro 'assert'
    (__builtin_expect(!(e), 0) ? __assert_rtn(__func__, __FILE__, __LINE__, #e) : (void)0)
                        ^
In file included from gevent-631/site-packages/gevent/__pycache__/_cffi__gd8408504xde6c861b.c:54:
In file included from /Users/rouge8/tmp/gevent-631/site-packages/gevent/libev.h:2:
/Users/rouge8/tmp/gevent-631/site-packages/gevent/libev/ev.c:4607:14: warning: expression result unused
      [-Wunused-value]
    assert (("libev: loop to be embedded is not embeddable", backend & ev_embeddable_backends ()));
             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/assert.h:93:25: note: expanded from macro 'assert'
    (__builtin_expect(!(e), 0) ? __assert_rtn(__func__, __FILE__, __LINE__, #e) : (void)0)
                        ^
30 warnings generated.
$ python ~/Desktop/631.py
<Response [200]>
$ pip uninstall -y gevent
Uninstalling gevent-1.1b2:
  Successfully uninstalled gevent-1.1b2
$ pip install gevent==1.1b2
Collecting gevent==1.1b2
Installing collected packages: gevent
Successfully installed gevent-1.1b2
$ python ~/Desktop/631.py
<Response [200]>
$ cat ~/Desktop/631.py
import gevent.monkey
gevent.monkey.patch_all()

import requests
print requests.get('https://www.google.com')
```

**Comment by rouge8** [comment]
Ah, somewhat less weird when I see that `pip uninstall gevent` leaves the compiled modules behind in the virtualenv's `site-packages`:

```console
$ tree gevent-631/site-packages/gevent/
gevent-631/site-packages/gevent/
└── __pycache__
    ├── _cffi__gd8408504xde6c861b.c
    ├── _cffi__gd8408504xde6c861b.pypy-26.so
    └── gevent-631
        └── site-packages
            └── gevent
                └── __pycache__
                    └── _cffi__gd8408504xde6c861b.o

5 directories, 3 files
```

**Comment by jamadden** [comment]
It's not clear to me why some systems see this and some don't; it seems that the initial installation sometimes fails to properly cache the compiled .so files, and I don't know why.

Fortunately the already-applied fix for #619 (9c5d4d524d08cb171f790a0e39f4493b6d80bb0e) applies here too. Thanks for the report.

**Comment by rouge8** [comment]
Weird. Thanks for the fast help!
